### PR TITLE
Avoid delegate allocations for ImmutableArray<T>.Any

### PR DESCRIPTION
**Pull request by sharwell**
This is the compiler code portion of #61719.

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Binder/SwitchBinder_Patterns.cs [hunk 1]
```diff
@@ -77,13 +77,13 @@ internal override BoundStatement BindSwitchStatementCore(SwitchStatementSyntax n
...
             var reachableLabels = decisionDag.ReachableLabels;
-            bool isSubsumed(BoundSwitchLabel switchLabel)
+            static bool isSubsumed(BoundSwitchLabel switchLabel, ImmutableHashSet<LabelSymbol> reachableLabels)
             {
                 return !reachableLabels.Contains(switchLabel.Label);
             }
 
             // If no switch sections are subsumed, just return
-            if (!switchSections.Any(s => s.SwitchLabels.Any(l => isSubsumed(l))))
+            if (!switchSections.Any(static (s, reachableLabels) => s.SwitchLabels.Any(static (l, reachableLabels) => isSubsumed(l, reachableLabels), reachableLabels), reachableLabels))
```
Comment (jcouv) [inline src/Compilers/CSharp/Portable/Binder/SwitchBinder_Patterns.cs:86]: This has become hard to follow (lots of nested parens)

**Review by jcouv — COMMENTED** [review]
Done with review pass (iteration 1). Only one minor concern.

File: src/Compilers/CSharp/Portable/Binder/SwitchBinder_Patterns.cs (hunk 1 above)
Comment (sharwell) [reply src/Compilers/CSharp/Portable/Binder/SwitchBinder_Patterns.cs:86]: I kept thinking a language feature supporting this case would be especially helpful.

Only a very small number of cases had a significant readability impact, so I tend to prefer leaving them for now and perhaps they can be refactored in the future. It would be nice to have analyzer, IDE, and/or language support to help with the scenario.

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 1)
